### PR TITLE
Feature/frontend setup

### PR DESCRIPTION
**Pull request by snaanortey**
Converted all js files to ts. Please review.

### CODE REVIEW

File: client/src/types.tsx [hunk 1]
```diff
@@ -0,0 +1,11 @@
+export interface Recipes  {
+
+    id: string;
+    imageUrl: string;
+    ingredients: string[];
+    picture_link?: string | null;
+    instructions: string;
+    title: string;
+
```
Comment (ehimah) [inline client/src/types.tsx:9]: clean up spaces

**Review by ehimah — CHANGES_REQUESTED** [review]
fix up

File: client/src/App.tsx [hunk 2]
```diff
@@ -1,24 +1,31 @@
...
+import NavBar from "./Components/NavBar/NavBar";
+import HomePage from "./Pages/HomePage/HomePage";
+import GetRecipes from "./Pages/GetRecipes/GetRecipes";
+import RecipeDetails from "./Pages/RecipeDetails/RecipeDetails";
+import { library } from "@fortawesome/fontawesome-svg-core";
+import { far } from "@fortawesome/free-regular-svg-icons";
+import GetShoppingList from "./Pages/GetShoppingList/GetShoppingList";
+import ShoppingList from "./Pages/ShoppingList/ShoppingList";
+
+// To migrate fontawesome icons
```
[the remark this answers is not in the document]
Comment (snaanortey) [reply client/src/App.tsx:13]: Change addressed

File: client/src/Components/NavBar/NavBar.scss [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+    margin: 0 16px;
+    color: var(--secoColor);
+    border-bottom: 2px solid rgba(0, 0, 0, 0.1);
+    text-decoration: none;
+    text-align: center;
+    transition: filter 200ms linear 0s;
+  }
+  .menu a:nth-of-type(1) {
+    // padding-top: 24px;
+  }
```
[the remark this answers is not in the document]
Comment (snaanortey) [reply client/src/Components/NavBar/NavBar.scss:85]: Addressed

File: server/package.json [hunk 4]
```diff
@@ -19,6 +19,10 @@
   },
   "homepage": "https://github.com/snaanortey/foodease#readme",
   "dependencies": {
+    "@fortawesome/fontawesome-svg-core": "^6.2.0",
+    "@fortawesome/free-regular-svg-icons": "^6.2.0",
+    "@fortawesome/free-solid-svg-icons": "^6.2.0",
+    "@fortawesome/react-fontawesome": "^0.2.0",
     "@elastic/elasticsearch": "^8.4.0",
```
[the remark this answers is not in the document]
Comment (snaanortey) [reply server/package.json:26]: Deleted them as discussed

File: client/src/types.tsx (hunk 1 above)
Comment (snaanortey) [reply client/src/types.tsx:9]: Updated

**Comment by snaanortey** [comment]
I have addressed all your comments. Thank you!

**Review by ehimah — APPROVED** [review]
🎉

**Pull request merged**